cmake/modules/arch_v2.cmake

```diff
@@ -30,6 +30,9 @@ if(HWMv2)
   # the Board / SoC path is no longer sufficient for determine the arch
   # (read: multi-core and multi-arch SoC).
   set(ARCH ${CONFIG_ARCH})
+  if(${CONFIG_ARCH} MATCHES "riscv")
```

Comment:
`riscv32` and `riscv64` share the same arch folder

Reply:
I guess this needs a better solution, @tejlmand ?

Reply:
not sure I follow what you try to do, but first line set `ARCH==${CONFIG_ARCH}`, so if `CONFIG_ARCH` is `riscv` then ARCH is also `riscv`.
Then in this `if` you match against `riscv` and sets `ARCH` to `riscv` if `CONFIG_ARCH` is `riscv`, but that was already the case in the line above.

Did you perhaps meant to do `if(${CONFIG_ARCH} MATCHES "riscv64")` ?

Reply:
what should be done here is to have add an extra entry here:
https://github.com/zephyrproject-rtos/zephyr/blob/c8a6ce58f03f9aa9f6678b7d974be55b912be237/arch/archs.yml#L14-L15
looking like this:
```
  - name: riscv64
    path: riscv
```

which will populate the `ARCH_V2_RISCV64_DIR` variable with the correct path, and thus things are expected to work.

------------


During the writing of this comment, then I noticed: https://github.com/zephyrproject-rtos/zephyr/blob/c8a6ce58f03f9aa9f6678b7d974be55b912be237/arch/riscv/Kconfig#L9-L10

`CONFIG_ARCH` is not named `riscv`, but `riscv32` or `riscv64`.
This is old mess, as Zephyr used to deduce the arch from path (which was named `riscv`), but we are changing to more proper way of doing this, which is using the Kconfig, named `riscv32` / `riscv64`.
So we should either update Kconfig to:
```
config ARCH
        string
        default "riscv64" if 64BIT
        default "riscv"
```
or update https://github.com/zephyrproject-rtos/zephyr/blob/c8a6ce58f03f9aa9f6678b7d974be55b912be237/arch/archs.yml#L14-L15
to
```
  - name: riscv32
    path: riscv
```

so that the arch names in use are aligned.